### PR TITLE
15757 ga events ch31

### PR DESCRIPTION
**Pull request by cohnjesse**
## Description
[Original Ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/11240)

This PR is to add Google Analytics events to the wizard for the chapter 31 form. A general spec for these events was provided in [this ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/15757). The spec is provided below as well as the corresponding screenshots of the events firing in dataSlayer for Firefox.

Some of the events in the spec have been moved into the orientation videos for which we do not have a full event spec yet. These events have been left out of this PR.

## Testing done
Tested events using dataSlayer for Firefox

## Screenshots

<details><summary>Form field changes (clicks on radio button answers)</summary>

Spec
```
'event': 'howToWizard-formChange',
'form-field-type': '<form_component_type>',
'form-field-label': <question_text>',
'form-field-value': '<question_answer_or_form_value>',
```

Actual event

Screen Shot 2020-12-09 at 11 08 58 AM

</details>

<details><summary>Received alert notice following a question answer</summary>

Spec
```
'event': 'howToWizard-alert-displayed',
'reason-for-alert': '<description>',
```

Actual event

Screen Shot 2020-12-09 at 11 09 57 AM

</details>

<details><summary>Click on link within blue box notice</summary>

Spec
```
'event': 'howToWizard-alert-link-click',
'howToWizard-alert-link-click-label'
```

Actual event

Screen Shot 2020-12-09 at 11 10 54 AM

</details>

<details><summary>User clicks link to skip how to wizard</summary>

Spec
```
'event': 'howToWizard-skip'
```

Actual event

Screen Shot 2020-12-09 at 11 11 25 AM

</details>

## Acceptance criteria
- [x] ode/event names provided by the GA/Insights Team has been inserted into our feature for tracking

### CODE REVIEW

File: src/applications/vre/28-1900/wizard/pages/helpers.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  recordEvent({
+    event: 'howToWizard-alert-displayed',
+    'reason-for-alert': description,
+  });
+};
+
+export const fireLinkClickEvent = event => {
+  recordEvent({
+    event: 'howToWizard-alert-link-click',
+    'howToWizard-alert-link-click-label': event.target.innerHTML,
```
Comment (micahchiang) [inline src/applications/vre/28-1900/wizard/pages/helpers.js:29]: would it be preferable to send `innerText` or `value`?

File: src/applications/vre/28-1900/wizard/pages/service-member/01-noHonorableDischarge.jsx [hunk 2]
```diff
@@ -1,14 +1,19 @@
 import React from 'react';
 import { serviceMemberPathPageNames } from '../pageList';
+import { recordNotificationEvent, fireLinkClickEvent } from '../helpers';
 
 const NoHonorableDischargeSM = () => {
+  recordNotificationEvent('ineligibility - recieved an honorable discharge');
```
Comment (micahchiang) [inline src/applications/vre/28-1900/wizard/pages/service-member/01-noHonorableDischarge.jsx:6]: recieved should be received

File: src/applications/vre/28-1900/wizard/pages/veteran/01-noHonorableDischarge.jsx [hunk 3]
```diff
@@ -1,14 +1,19 @@
 import React from 'react';
 import { veteranPathPageNames } from '../pageList';
+import { recordNotificationEvent, fireLinkClickEvent } from '../helpers';
 
 const NoHonorableDischarge = () => {
+  recordNotificationEvent('ineligibility - recieved an honorable discharge');
```
Comment (micahchiang) [inline src/applications/vre/28-1900/wizard/pages/veteran/01-noHonorableDischarge.jsx:6]: recieved should be received.

**Review by micahchiang — APPROVED** [review]
lgtm just some spelling fixes.

File: src/applications/vre/28-1900/wizard/pages/helpers.js (hunk 1 above)
Comment (cohnjesse) [reply src/applications/vre/28-1900/wizard/pages/helpers.js:29]: They all send the same thing. Updated this to `innerText`.

File: src/applications/vre/28-1900/wizard/pages/service-member/01-noHonorableDischarge.jsx (hunk 2 above)
Comment (cohnjesse) [reply src/applications/vre/28-1900/wizard/pages/service-member/01-noHonorableDischarge.jsx:6]: Updated

File: src/applications/vre/28-1900/wizard/pages/veteran/01-noHonorableDischarge.jsx (hunk 3 above)
Comment (cohnjesse) [reply src/applications/vre/28-1900/wizard/pages/veteran/01-noHonorableDischarge.jsx:6]: Updated

**Pull request merged**